src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -805,6 +805,21 @@ public static void firePlayerChangedDimensionEvent(Player player, ResourceKey<Le
         MinecraftForge.EVENT_BUS.post(new PlayerEvent.PlayerChangedDimensionEvent(player, fromDim, toDim));
     }
 
+    public static boolean firePrePlayerLoggedIn(Player player, Connection connection)
+    {
+        PlayerEvent.PrePlayerLoginEvent event = new PlayerEvent.PrePlayerLoginEvent(player, connection);
+        MinecraftForge.EVENT_BUS.post(event);
+
+        if (event.isCanceled())
+        {
+            connection.send(new ClientboundDisconnectPacket(event.getDenyMessage()));
+            connection.disconnect(event.getDenyMessage());
```

Comment:
What's the reason for not using `ServerLoginPacketListenerImpl#disconnect`?

Reply:
I initially attempted to use that method and got a buffer overflow exception on the client end. Couldn't figure out why it wasn't working (https://i.gyazo.com/78681e94197d139d2c63dfdd9c2e3508.png). I believe it's something to do with when the event is fired (too late to use this method?)

Reply:
The method literally does the same thing, it just has a try-catch around it. So that sounds improbable.

Reply:
Actually not quite. The method in ServerLoginPacketListenerImpl uses a "ClientboundLoginDisconnectPacket" whereas I'm using a "ClientboundDisconnectPacket"